sdk/core/Azure.Core/samples/Mocking.md

```diff
@@ -1,55 +1,3 @@
 # Azure.Core mocking samples
 
-**NOTE:** Samples in this file apply only to packages that follow [Azure SDK Design Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html). Names of such packages usually start with `Azure`. 
-
-## Creating a mock of the client using Moq
-
-You can use a combination of model factory and `Mock` class to create a mock of a client:
-
-```C# Snippet:ClientMock
-// Create a mock response
-var mockResponse = new Mock<Response>();
-
-// Create a mock value
-var mockValue = SecretModelFactory.KeyVaultSecret(
-    SecretModelFactory.SecretProperties(new Uri("http://example.com"))
-);
-
-// Create a client mock
-var mock = new Mock<SecretClient>();
-
-// Setup client method
-mock.Setup(c => c.GetSecret("Name", null, default))
-    .Returns(Response.FromValue(mockValue, mockResponse.Object));
-
-// Use the client mock
-SecretClient client = mock.Object;
-KeyVaultSecret secret = client.GetSecret("Name");
-```
-
-## Creating a mock of the method that returns Pageable
-
-For methods that return instances of `Pageable` or `AsyncPageable`, `[Async]Pageable<T>.FromPages` method can be used to create an instance for test:
-
-```C# Snippet:ClientMockWithPageable
-// Create a client mock
-var mock = new Mock<SecretClient>();
-
-// Create a Page
-var deletedValue = SecretModelFactory.DeletedSecret(
-    SecretModelFactory.SecretProperties(new Uri("http://example.com"))
-);
-var pageValues = new[] { deletedValue };
-var page = Page<DeletedSecret>.FromValues(pageValues, default, new Mock<Response>().Object);
-
-// Create a mock for the Pageable
-var pageable = Pageable<DeletedSecret>.FromPages(new[] { page });
-
-// Setup client method that returns Pageable
-mock.Setup(c => c.GetDeletedSecrets(default))
-    .Returns(pageable);
-
-// Use the client mock
-SecretClient client = mock.Object;
-DeletedSecret deletedSecret = client.GetDeletedSecrets().First();
-```
+This guidance has moved to [Unit testing and mocking with the Azure SDK for .NET](https://learn.microsoft.com/dotnet/azure/sdk/unit-testing-mocking).
```

Comment:
Curious: why wouldn't we keep some mocking samples in the GH repo?  Is this part of a larger plan to move core guidance out of GH into learn docs?  If the latter, could you share more information on this?

Reply:
Sounds good.  As discussed offline, I 100% support having great docs on Learn, and if eventually we migrate docs updates to that, as long as we understand the process and know how to do it, that is all goodness.  In the meantime, it might make sense to leave Core docs and samples in the GH repo so we can improve/update them at the same time as changes go in to the types -- I wouldn't want us to stop keeping docs updated just because of process overhead or confusion about where to document changes.  Maybe we could have the docs/Learn team store commit hashes for article updates so they can get a quick sense of whether the GH docs have changed since they wrote an article?  And it's possible we think that mocking won't change in the short term, so it's ok to delete these docs because all the process questions will get resolved before the next round of changes here?  Overall, I think it would be good to understand the plan so we know what the implications are for updating GH docs going forward.

Reply:
The concern there is that we're duplicating content and potentially introducing drift.   Then, we have two different sets of guidance and customers now need to figure out which is the "right" one.   In this case, the learn sample is much more comprehensive than the Core sample was and is a better developer experience.   

I'd prefer to see us contribute and review updates to the learn article than maintain our own separate copy of things.  In my mind, ensuring that we're invested and involved in these articles is the key to them being successful.  There are too many today that we were not involved with and which have deviated from our recommended patterns and guidance.

Reply:
Cool, I also like not having duplication, as long as we don't run into process issues around making updates as they happen - we are already slow to update docs in some places, and I wouldn't want the docs to get out of sync with the code because we couldn't put in a change as part of the same PR (for example).  As long as the process is clear and doable, I am fine with any option.   My primary concern is that customers don't have accurate docs in any place that reflect the code they're using.

Reply:
The docs process is just like any other pull request.  You can get to the GH source for an article using the button on the bottom of the page.   It is advised that you move to the `-pr` repo for submitting pull requests as an internal.  Otherwise, it's just an edit-and-PR.

Reply:
One small correction to Jesse's statement above: The private mirror repo doesn't exist for dotnet/docs. All contributions are to be made publicly. Their contributions guidelines are already documented at https://learn.microsoft.com/contribute/dotnet/dotnet-contribute.